### PR TITLE
Refactor OIDC siop wallet

### PR DESCRIPTION
**Pull request by nodh**
Diff may looks huge, but the single commits are rather small

### CODE REVIEW

File: vclib-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidc/OidcSiopWallet.kt [hunk 1]
```diff
@@ -270,139 +271,103 @@ class OidcSiopWallet(
...
-                            .also { Napier.w("client_id_scheme is $clientIdScheme, but client_id does not match redirect_uri") }
-                }
-            }
+        if (clientIdScheme.isAnyX509()) {
+            runCatching { params.verifyClientIdSchemeX509() }
+                .onFailure { return KmmResult.failure(it) }
         }
 
         // params.clientIdScheme is assumed to be OpenIdConstants.ClientIdSchemes.REDIRECT_URI,
         // because we'll require clientMetadata to be present, below
```
Comment (acrusage-iaik) [inline vclib-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidc/OidcSiopWallet.kt:289]: Probably a good time to also remove this comment

File: vclib-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidc/OidcSiopWallet.kt [hunk 2]
```diff
@@ -270,139 +271,103 @@ class OidcSiopWallet(
...
-                && clientMetadata.vpFormats.jwtSd?.algorithms?.contains(jwsService.algorithm.identifier) != true
-            ) return KmmResult.failure<AuthenticationResponseParameters>(OAuth2Exception(Errors.REGISTRATION_VALUE_NOT_SUPPORTED))
-                .also { Napier.w("Incompatible JWT algorithms") }
-            if (clientMetadata.vpFormats.msoMdoc?.algorithms != null
-                && clientMetadata.vpFormats.msoMdoc?.algorithms?.contains(jwsService.algorithm.identifier) != true
-            ) return KmmResult.failure<AuthenticationResponseParameters>(OAuth2Exception(Errors.REGISTRATION_VALUE_NOT_SUPPORTED))
-                .also { Napier.w("Incompatible JWT algorithms") }
+            runCatching { clientMetadata.vpFormats.verifySupport(jwsService.algorithm) }
+                .onFailure { return KmmResult.failure(it) }
+        }
```
Comment (acrusage-iaik) [inline vclib-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidc/OidcSiopWallet.kt:310]: This check is probably obsolete, the formats of the submitted tokens are validated at lines 435-458 of the old file version (line 319-324 of commit 88cd53f3017a6e93dfc23604f64fd095838e33e7). 

Reason: It's not important whether the relying part supports a format if no token of that format is issued

Blame: https://github.com/a-sit-plus/kmm-vc-library/commit/27da00a2402bba8d7fc51544449160fe9b361892#r142288803

**Review by acrusage-iaik — CHANGES_REQUESTED** [review]
Looks good, but maybe remove the obsolete format validation.

File: vclib-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidc/OidcSiopWallet.kt (hunk 2 above)
Comment (nodh) [reply vclib-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidc/OidcSiopWallet.kt]: Done in https://github.com/a-sit-plus/kmm-vc-library/pull/68/commits/5cda85bcada80db90bdd352ba0092b1c3c0553a9

File: vclib-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidc/OidcSiopWallet.kt (hunk 1 above)
Comment (nodh) [reply vclib-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidc/OidcSiopWallet.kt]: Done in https://github.com/a-sit-plus/kmm-vc-library/pull/68/commits/5cda85bcada80db90bdd352ba0092b1c3c0553a9

**Review by acrusage-iaik — APPROVED** [review]

**Pull request merged**